src/main/scala/introprog/PixelWindow.scala

```diff
@@ -254,6 +254,19 @@ class PixelWindow(
     Swing.await { new java.awt.Color(canvas.img.getRGB(x, y)) }
   }
 
+  import java.awt.image.BufferedImage
+  /**
+    * Returns a screenshot of the window
+    * SLOW! creates image one pixel at a time
+    */
+  def getImage(): BufferedImage = {
+    val img = new BufferedImage(width, height, BufferedImage.TYPE_INT_RGB);
+    (0 until width).map(w => (0 until height).map(h => 
```

Comment:
should be foreach instead of map. Or even better, use Scala 3-syntax with nested `for ... do`.